makefiles/dependency_resolution.inc.mk

```diff
@@ -2,6 +2,10 @@
 # until no new modules, pkgs, or features are pull in order to catch all
 # transient dependencies
 
+# Locate used modules in $(EXTERNAL_MODULE_DIRS).
+EXTERNAL_MODULE_PATHS := $(sort $(foreach dir,$(EXTERNAL_MODULE_DIRS),\
+  $(foreach mod,$(USEMODULE),$(dir $(wildcard $(dir)/$(mod)/Makefile)))))
+
```

Comment:
Re-looking at the code, and comment/question in https://github.com/RIOT-OS/RIOT/pull/16104#issuecomment-806163303, this would mean all `EXTERNAL_MODULE_DIRS` would need to be absolute paths no?